File: component/CommonComponent/src/main/java/com/energy/commoncomponent/bean/RotateDegree.java
package com.energy.commoncomponent.bean;

/**
 * Created by fengjibo on 2023/7/4.
 */
public enum RotateDegree {

    DEGREE_0(0),
    //
    DEGREE_90(1),
    //
    DEGREE_180(2),
    //
    DEGREE_270(3);

    /**
     * Private method description.
     */
    private final int value;

    RotateDegree(int value) {
        this.value = value;
    }

    /**
     * Method description.
     */
    public int getValue() {
        return value;
    }

    /**
     * Method description.
     */
    public static RotateDegree valueOf(int value) {
        RotateDegree[] types = RotateDegree.values();
        for(RotateDegree type: types){
            if (type.value == value) {
                return type;
            }
        }
        return DEGREE_0;
    }
}

File: component/thermal-ir/src/main/java/com/energy/commoncomponent/bean/RotateDegree.kt
package com.energy.commoncomponent.bean

/**
 * Temporary RotateDegree enum to fix compilation issues.
 * This should be replaced with the proper external library dependency.
 */
enum class RotateDegree(private val value: Int) {
    DEGREE_0(0),
    DEGREE_90(90),
    DEGREE_180(180),
    DEGREE_270(270);

    fun getValue(): Int = value
}